rootgrow/rootgrow.py

```diff
@@ -97,7 +97,9 @@ def get_partition_id_from_root(disk_device, root_device):
                 for c in reversed(device):
                     if c.isdigit():
                         partition_index = c + partition_index
-                return int(partition_index)
+                    else:
+                        return int(partition_index)
```

Comment:
This is very confusing to me and also seems wrong. In the reversed lookup of devices why do you return the accumulated number of the partition id if you don't see a digit anymore ?

From my perspective this function needs to be refactored. I suggest the following

1. Change the lsblk call as follows
  
    ```
    lsblk -p -l -o NAME,TYPE disk_device
    ```

    This sequence of options is safe across all distros and does not expose the ordering issue
    (we use it in kiwi for several months and have tested it through the available distros)

     The output looks like this example

     ```
     $ lsblk -p -l -o NAME,TYPE /dev/sda

     NAME      TYPE
     /dev/sda  disk
     /dev/sda1 part
     /dev/sda2 part
     /dev/sda3 part
     /dev/sda4 part
     ```

2. Let the function walk through this output (tested by unit testing) and
    calculate on the number of available "part" elements. Any line that is typed
    with "part" or "md" is a relevant selection to calculate as a partition

The way this is done here is confusing and the mocked lsblk output is not something I can
imagine to come from lsblk in real live.

I strongly recommend to make this more clear and align the testing

Reply:
I think the accumulation is to handle if partitions > 9.

Reply:
The only thing this PR changes is that when determining the partition ID (assuming a numeric ID), it stops at the first non-digit while traversing the string backwards, rather than running through the whole string like before, which is the reason why it returned a partition ID of `13` for a device name like `/dev/nvme0n1p3`.

Whether the implementation should be different is another conversation IMHO, which probably should have taken place before the last merge. I don't understand why `get_partition_id_from_root` even calls `lsblk` again to be honest.

But as said, this simply fixes a bug that was introduced by https://github.com/SUSE-Enceladus/rootgrow/pull/7.

Reply:
Agreed, since this is now tested and looks good I think it's good to move forward. And make a note for follow on PR to clean things up.

Reply:
ok, with Joe adding more details on what lead to this change I'm fine if you move forward. But seriously this code should be refactored.

Reply:
Thanks for finding and fixing the logic error introduced in #7. The point of #7 was that the order of what lsblk outputs is no longer in the order we expected to begin with or we would not have had issue with expansion to begin with. The previous implementation assumed that lsblk produces ordered output.